posts/2021-05-14-ldap-multipart-21005.adoc

```diff
@@ -0,0 +1,225 @@
+
+To update an `LdapRegistry` to use the GSSAPI/Kerberos option, you can set the bind authentication mechanism type using the new `LdapRegistry` attribute, `bindAuthMechanism`:
+
+[source, xml]
+----
+bindAuthMechanism="GSSAPI"
+----
+
+You also need the Kerberos principal or Service Principal Name:
+
+[source, xml]
+----
+krb5Principal="user1@EXAMPLE.COM"
+----
+
+If you are using a Kerberos credential cache (ticket cache or ccache) to store the Kerberos credentials, add the ticket cache file name to the `LdapRegistry` with the new attribute, `krb5TicketCache`:
+
+[source, xml]
+----
+krb5TicketCache="${server.config.dir}/security/krb5-user1.cc"
+----
+
+If you are using a custom Kerberos configuration file (krb.conf or krb.ini), set the file name using the global Kerberos configuration element:
+
+[source, xml]
+----
+<kerberos configFile="${server.config.dir}/security/krb5.conf"/>
+----
+
+If you are using a Kerberos keytab file to store encrypted keys for principals, set the file using the global Kerberos configuration element:
+
+[source, xml]
+----
+<kerberos keytab="${server.config.dir}/security/krb5.keytab" configFile="${server.config.dir}/security/krb5.conf"/>
+----
+
+If the Kerberos configuration file is not defined, Open Liberty will attempt to resolve by using the JDK default locations and the operating system default locations.
+
+For the Kerberos credentials, the locations are checked in the following order: the ticket cache (if provided), the configured keytab file, and finally the JDK default location.
+
+The following example shows how to configure the `LdapRegistry` element using a ticket cache and custom Kerberos config file:
+
+[source, xml]
+----
+<kerberos keytab= configFile="${server.config.dir}/security/krb5.conf"/>
+
+<ldapRegistry id="LDAP" realm="SampleLdapADRealm" host="ldap_hostname" port="389" ignoreCase="true" baseDN="DC=example,DC=com" bindAuthMechanism="GSSAPI" krb5Principal="user1@EXAMPLE.COM" krb5TicketCache="${server.config.dir}/security/krb5-user1.cc" ldapType="Custom" />
+----
+
+The following example shows how to configure an LDAP Registry using a keytab and custom Kerberos config file:
+
+[source, xml]
+----
+<kerberos keytab="${server.config.dir}/security/krb5.keytab" configFile="${server.config.dir}/security/krb5.conf" />
+
+<ldapRegistry id="LDAP" realm="SampleLdapADRealm" host="ldap_hostname" port="389" ignoreCase="true" baseDN="DC=example,DC=com" bindAuthMechanism="GSSAPI" krb5Principal="user1@EXAMPLE.COM" ldapType="Custom" />
+----
+
+For more information on `LdapRegistry`, see the link:https://openliberty.io/docs/latest/reference/feature/ldapRegistry-3.0.html[LDAP User Registry documentation].
+
+To enable this new beta function in your app, pull the All Beta Features package and add the LDAP User Registry 3.0 feature to your `server.xml` file:
```

Comment:
This is no longer beta.
"To enable this function, add the LDAP User Registry 3.0 feature to your `server.xml` file:"

Reply:
Updated.